TradeBeacons/data/scripts/systems/tradingoverview.lua

```diff
@@ -102,28 +102,64 @@ function getTradeBeaconScanRange(seed, rarity)
 end
 
 function requestSectorsData(caller)
-    if not caller then
+    local player
+    if caller then
+        -- triggered by player interaction so we can be specific
+        player = Player(caller)
+    elseif onServer() then
+        local faction = getParentFaction()
```

Comment:
this is a similar to how trade beacons are registering data:
get the parent faction, if it's a player, that's the player to register with (what if that player isn't on line?)
if it's an alliance, the beacon notifies all players, here we just need one of the players. this will be a little bit odd if the player we choose has extra, non-alliance trade beacons too. if that happens then this server-side call will get that trade data as well as the alliance trade data.